tests/PactBuilder/JsonPatternBuilderTest.php

```diff
@@ -17,87 +18,231 @@ public function testItBuildsMessage(): void
+            ],
+        ])->toArray(), $body->matchingRules()->toArray());
+    }
+
+    public function testItGeneratesWithPaths(): void
+    {
+        $builder = new JsonPatternBuilder();
+        $builder = $builder->withPattern([
+            'array' => [
+                $builder->like(1),
+                $builder->like(2),
+            ],
+            '.' => [
+                '[' => $builder->like(1),
+                ']' => $builder->like(1),
+            ],
+            'object' => [
+                '(' => $builder->like(1),
+                ')' => $builder->like(1),
+            ],
+            '"' => $builder->like(1),
+            "'" => $builder->like(1),
+        ]);
+
+        $body = $builder->build();
+
+        Assert::assertSame(\json_encode([
+            'array' => [
+                1,
+                2
+            ],
+            '.' => [
+                '[' => 1,
+                ']' => 1,
+            ],
+            'object' => [
+                '(' => 1,
+                ')' => 1,
+            ],
+            '"' => 1,
+            "'" => 1,
+        ], \JSON_THROW_ON_ERROR), $body->content());
+
+        Assert::assertSame(MatchingRules::fromArray([
+            "$.array[0]" => [
+                'matchers' => [
+                    ['type' => 'type'],
+                ],
+            ],
+            '$.array[1]' => [
                 'matchers' => [
                     ['type' => 'type'],
                 ],
             ],
-            '$.deep_type.some' => [
-                'combine' => 'AND',
+            "$['.']['[']" => [
                 'matchers' => [
-                    ['type' => 'regex', 'pattern' => '/.*/'],
+                    ['type' => 'type'],
                 ],
             ],
-            '$.some' => [
-                'combine' => 'AND',
+            "$['.'][']']" => [
                 'matchers' => [
-                    ['type' => 'regex', 'pattern' => '/.*/'],
+                    ['type' => 'type'],
                 ],
             ],
-            '$.another.deep' => [
-                'combine' => 'AND',
+            "$.object['(']" => [
                 'matchers' => [
-                    ['type' => 'regex', 'pattern' => '/.*/'],
+                    ['type' => 'type'],
                 ],
             ],
-            '$.array[0]' => [
-                'combine' => 'AND',
+            "$.object[')']" => [
                 'matchers' => [
-                    ['type' => 'regex', 'pattern' => '/.*/'],
+                    ['type' => 'type'],
                 ],
             ],
-        ], $body->matchingRules()->toArray());
-    }
-
-    public function testItIsImmutable(): void
-    {
-        $builder = new JsonPatternBuilder();
-
-        Assert::assertNotSame($builder, $builder->withPattern([]));
+            "$['\"']" => [
+                'matchers' => [
+                    ['type' => 'type'],
+                ],
+            ],
+            "$['\'']" => [
```

Comment:
It's really hard to find out how this should be escaped - the tool at http://jsonpath.com/ wants `'` to be unescaped.